CMakeLists.txt

```diff
@@ -57,6 +57,15 @@ option(USE_SNPE "Use Qualcomm's SNPE library" OFF)
 option(USE_THREADS "Use Threads" ON)
 option(USE_ZMQ "Use ZMQ" OFF)
 
+# Remove CMAKE_INSTALL_PREFIX from CMAKE_SYSTEM_PREFIX_PATH
+# Otherwise, third_party libraries will be found there for rebuilds
+# Only remove one copy in case the install prefix is /usr/local
+list(FIND CMAKE_SYSTEM_PREFIX_PATH "${CMAKE_INSTALL_PREFIX}" found)
+if(NOT found EQUAL -1)
+    list(REMOVE_AT CMAKE_SYSTEM_PREFIX_PATH ${found})
```

Comment:
I'd quite like it if cmake told me that it was futzing with my CMAKE_SYSTEM_PREFIX_PATH; it would probably save me some debugging if it was triggering when I didn't expect it to!

Reply:
Yes, a `message()` is appropriate here, I agree.

Reply:
Also, I'm not convinced this is exactly right. If the install prefix is `/usr/local` and the default value for CMAKE_SYSTEM_PREFIX_PATH is `/usr/local;/usr;/;/usr;/usr/local`, then this will remove the first `/usr/local` (standard search path) instead of the last one (install prefix). That will mess with the ordering in a weird way. But I'm not sure we can hard-code removing the 4th value for all CMake versions.

Reply:
You can use [CMAKE_FIND_NO_INSTALL_PREFIX](https://cmake.org/cmake/help/v3.9/variable/CMAKE_FIND_NO_INSTALL_PREFIX.html) to tell CMake not to add the install prefix to the search path in the first place.

Reply:
@bradking thanks for chiming in, but that doesn't seem to work, either. The docs say that variable only affects `find_package()`, but we're calling `find_path()` and `find_library()`: https://github.com/caffe2/caffe2/blob/v0.8.1/cmake/Modules/FindBenchmark.cmake

Reply:
It needs to be set very early, such as before the first `project` or `enable_language` command, in order to influence the initialization of `CMAKE_SYSTEM_PREFIX_PATH`.

Reply:
Oh wow, that's early. Thanks for the tip - that works!